code/class/registry/registry.php

```diff
@@ -129,7 +136,54 @@ public function alias($class, $alias)
     }
 
     /**
-     * Get a list of all the identifier aliases
+     * Get a specific locator from the class namespace
+     *
+     * @return string|false The name of the locator or FALSE if no locator could be found
+     */
+    public function getLocator($class)
+    {
+        $result = false;
+
+        if(!false == $pos = strrpos($class, '\\'))
+        {
+            $namespace = substr($class, 0, $pos);
+
+            if(isset($this->_namespaces[$namespace])) {
+                $result = $this->_namespaces[$namespace];
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * Set a specific locator based on a class namespace
+     *
+     * @return  ClassRegistry
+     */
+    public function setLocator($class, $locator)
+    {
+        if(!false == $pos = strrpos($class, '\\'))
```

Comment:
Additionally, per the code, you can only ever have 1 locator registered to each namespace within the registry, yet you can register multiple locators within the `ClassLoader`. Should it be a namespace => loactors map, rather than locator.